wsmaster/che-core-api-factory/src/test/java/org/eclipse/che/api/factory/server/urlfactory/DefaultFactoryUrlTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2012-2023 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.factory.server.urlfactory;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+/** Testing {@link DefaultFactoryUrl} */
+@Listeners(MockitoTestNGListener.class)
+public class DefaultFactoryUrlTest {
+  @Test
+  public void shouldCredentials() {
+    // given
+    DefaultFactoryUrl factoryUrl =
+        new DefaultFactoryUrl().withUrl("https://username:password@hostname/path");
```

Comment:
could we add  some more specific unit tests to cover more potential edge cases e.g. we can use specific URLs like the one provided in the issue description for azure https://github.com/eclipse/che/issues/21998
```
https://mrebmvdxzlhhmratfm5iqzf5ox4iagdc2zhgcwrhbref7cihte6q@dev.azure.com/marioloriedo/marioloriedo/_apis/sourceProviders/TfsGit/filecontents?repository=private-repo&commitOrBranch=main&path=/.devfile.yaml&api-version=7.0
```
would be nice to cover also github, gitlab and bitbuket e.g. sample for GitHub:

```
https://$GH_PAT@raw.githubusercontent.com/<user or organization>/<repo name>/<branch>/.devfile.yaml
````

Reply:
done